--- a/components/sdmmc/sdmmc_cmd.c	2025-12-14 12:56:05.965876398 +0100
+++ a/components/sdmmc/sdmmc_cmd.c	2025-12-14 13:14:25.211350030 +0100
@@ -20,7 +20,12 @@
     int slot = card->host.slot;
     ESP_LOGV(TAG, "sending cmd slot=%d op=%d arg=%x flags=%x data=%p blklen=%d datalen=%d timeout=%d",
             slot, cmd->opcode, cmd->arg, cmd->flags, cmd->data, cmd->blklen, cmd->datalen, cmd->timeout_ms);
-    esp_err_t err = (*card->host.do_transaction)(slot, cmd);
+    esp_err_t err = 0;
+    unsigned char retries = 0;
+    do {
+      retries++;
+      err = (*card->host.do_transaction)(slot, cmd);
+    } while ((retries < 5) && (err != 0));
     if (err != 0) {
         ESP_LOGD(TAG, "cmd=%d, sdmmc_req_run returned 0x%x", cmd->opcode, err);
         return err;
